alf/algorithms/taac_algorithm.py

```diff
@@ -475,13 +475,14 @@ def _predict_action(self,
                         mode=Mode.rollout):
 
         observation = time_step.observation
+        obs_shape = alf.nest.get_nest_shape(observation)
```

Comment:
Why can you assume all the fields of observation have the same shape?

Reply:
> Why can you assume all the fields of observation have the same shape?

Here I only need to extract the shape from an arbitrary element, because later I only use its batch dim.

Reply:
If it's for batch_size, then you can use `batch_shape = time_step.step_type.shape`